editor/src/clj/editor/scene.clj

```diff
@@ -1051,21 +1049,74 @@
+        delta (c/camera-unproject camera-b viewport (.x point) (.y point) (.z point))]
+    (.sub delta world)
+    (c/camera-move camera-a (.x delta) (.y delta) (.z delta))))
+
+(defn- get-3d-camera
+  [camera]
+  (or (g/node-value camera :cached-3d-camera)
+      (c/tumble (g/node-value camera :local-camera) 200.0 -100.0)))
+
+(defn- camera-2d?
+  [view]
+  (some-> (view->camera view)
+          (g/node-value :local-camera)
+          (c/mode-2d?)))
+
+(defmulti realign-camera (fn [view _animate?] (if (camera-2d? view) :2d :3d)))
+
+(defmethod realign-camera :2d
+  [view animate?]
+  (let [camera-node (view->camera view)
+        local-cam (g/node-value camera-node :local-camera)
+        viewport (g/node-value view :viewport)
+        camera-3d (-> (get-3d-camera camera-node)
+                      (sync-camera-position local-cam viewport))
+        local-cam (cond-> local-cam
+                    (= (:type camera-3d) :perspective)
+                    (c/camera-orthographic->perspective c/fov-y-35mm-full-frame))]
+    (set-camera! camera-node local-cam camera-3d animate?)))
+
+(defmethod realign-camera :3d
+  [view animate?]
+  (let [camera-node (view->camera view)
+        local-cam (g/node-value camera-node :local-camera)
+        is-perspective (= (:type local-cam) :perspective)]
+    (g/transact (g/set-property camera-node :cached-3d-camera local-cam))
+    (let [end-camera (cond-> local-cam
+                       is-perspective c/camera-perspective->orthographic
+                       :always c/camera-orthographic-realign
+                       is-perspective (c/camera-orthographic->perspective c/fov-y-35mm-full-frame))]
+      (set-camera! camera-node local-cam end-camera animate? #(set-camera-type! view :orthographic)))))
+
 (handler/defhandler :frame-selection :global
   (active? [app-view evaluation-context]
            (active-scene-view app-view evaluation-context))
   (enabled? [app-view evaluation-context]
             (when-let [view (active-scene-view app-view evaluation-context)]
               (let [selected (g/node-value view :selection evaluation-context)]
                 (not (empty? selected)))))
-  (run [app-view] (when-let [view (active-scene-view app-view)]
-                    (frame-selection view true))))
+  (run [app-view] (some-> (active-scene-view app-view)
+                          (frame-selection true))))
+
+(defn- camera-animating?
+  [app-view]
+  (some-> (active-scene-view app-view)
+          (view->camera)
+          (g/node-value :animating)))
 
 (handler/defhandler :realign-camera :global
   (active? [app-view evaluation-context]
            (active-scene-view app-view evaluation-context))
-  (run [app-view] (when-let [view (active-scene-view app-view)]
-                    (realign-camera view true))))
+  (enabled? [app-view] (not (camera-animating? app-view)))
+  (run [app-view] (some-> (active-scene-view app-view) 
+                          (realign-camera true))))
```

Comment:
`:realign-camera` is probably not a good name anymore, but I didn't want to add a new shortcut. Using `period` to toggle the camera is fine in my opinion, and migrating to a new key seems like an overkill.